app/src/main/java/org/mozilla/focus/tabs/tabtray/TabTrayFragment.java

```diff
@@ -251,6 +250,9 @@ public void closeTabTray() {
 
     @Override
     public void navigateToHome() {
+        if (isOpenedByHome()) {
```

Comment:
As discussed, steps to improve this would be
1. Make geo dialog a view instead of AlertDialog
2. Remove TabTrayFragment.FRAGMENT_TAG from MainMediator#FRAGMENTS_SEQUENCE
3. Move this protection stuff from TabTrayFragment to ScreenNavigator#addHomeScreen()